addon/actions/action.js

```diff
@@ -80,11 +80,13 @@ export default Object.extend({
 
   _promise() {
     return this.get('adapter').ajax(this.get('url'), this.get('requestType'), this.get('data')).then((response) => {
-      if (this.get('config.pushToStore') && response.data) {
+      if (this.get('config.pushToStore') && response) {
         return this.get('serializer').pushPayload(this.get('store'), response);
       } else {
         return response;
       }
+    }).catch((errors) => {
+      return this.get('serializer').extractErrors(this.get('store'), this.get('store').modelFor(this.get('modelName')), errors, this.get('model.id'));
```

Comment:
Thank you for your comments ! :) I made the updates. Does this sound good to you ? I just didn't figure out the rethrow thing.